typescript/sdk/src/middleware/account/InterchainAccount.ts

```diff
@@ -56,4 +74,125 @@ export class InterchainAccount extends RouterApp<InterchainAccountFactories> {
     );
     return new InterchainAccount(helper.contractsMap, helper.multiProvider);
   }
+
+  async deployAccount(
+    chain: ChainName,
+    config: AccountConfig,
+    routerOverride?: Address,
+    ismOverride?: Address,
+  ): Promise<Address> {
+    const originDomain = this.multiProvider.tryGetDomainId(config.origin);
+    if (!originDomain) {
+      throw new Error(
+        `Origin chain (${config.origin}) metadata needed for deploying ICAs ...`,
+      );
+    }
+    const localRouter = this.router(this.contractsMap[chain]);
+    const routerAddress =
+      routerOverride ??
+      bytes32ToAddress(await localRouter.routers(originDomain));
+    const ismAddress =
+      ismOverride ?? bytes32ToAddress(await localRouter.isms(originDomain));
+    const account = await localRouter[
+      'getLocalInterchainAccount(uint32,address,address,address)'
+    ](originDomain, config.owner, routerAddress, ismAddress);
+    if (
+      (await this.multiProvider.getProvider(chain).getCode(account)) === '0x'
```

Comment:
Nit, some reusable util for checking if an address is/isn't a deployed contract would be good